staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -300,14 +304,22 @@ func (flags *ApplyFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, baseNa
 
 	var applySet *ApplySet
 	if flags.ApplySetRef != "" {
-		var applySetNs string
+		parent, err := ParseApplySetParentRef(flags.ApplySetRef, mapper)
+		if err != nil {
+			return nil, fmt.Errorf("invalid parent reference %q: %w", flags.ApplySetRef, err)
+		}
 		// ApplySet uses the namespace value from the flag, but not from the kubeconfig or defaults
-		if enforceNamespace {
-			applySetNs = namespace
+		// This means the namespace flag is required when using a namespaced parent.
+		if enforceNamespace && parent.IsNamespaced() {
+			parent.Namespace = namespace
 		}
-		if applySet, err = NewApplySet(flags.ApplySetRef, applySetNs, mapper); err != nil {
-			return nil, err
+		// TODO: is version.Get() the right thing? Does it work for non-kubectl package consumers?
```

Comment:
TODO is probably obsolete now (or should be moved) :-)

Reply:
You're right, it should go! I'll do it in a follow up, assuming the tests pass.